packages/client-config/src/client-config-contributor/client_config_contributor_v1.ts

```diff
@@ -258,10 +258,17 @@ export class StorageClientConfigContributor implements ClientConfigContributor {
       return {};
     }
     const config: Partial<clientConfigTypesV1.AWSAmplifyBackendOutputs> = {};
-
+    const bucketsStringArray = JSON.parse(
+      storageOutput.payload.buckets ?? '[]'
+    );
     config.storage = {
       aws_region: storageOutput.payload.storageRegion,
       bucket_name: storageOutput.payload.bucketName,
+      buckets: bucketsStringArray.map((b: string) => ({
+        name: JSON.parse(b).name,
+        bucket_name: JSON.parse(b).bucketName,
+        aws_region: JSON.parse(b).storageRegion,
+      })),
```

Comment:
we should validate that the parsed elements are actually the expected shape. We can also simplify a bit to only parse once:

```suggestion
      buckets: bucketsStringArray
        .map((bucketJsonString) => zodSchema.parse(JSON.parse(bucketJsonString))
        .map(({name, bucketName, storageRegion}) => ({
          name,
          bucket_name: bucketName,
          aws_region: storageRegion
        })
```